### ISSUE TITLE
Heading level violation on Benefits hub template

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#12169]
**1/18: blocked on content for new H2 – requested SWContent input**

## Describe the defect
On Benefits Hub Landing pages (at minimum) there is an accessibility error because the heading structure is not correct.

## To Reproduce
1. Go to [Disability Benefits](https://www.va.gov/disability/)
2. Run the axe browser plugin
3. Note that an error is flagged indicating heading levels are skipped:
![image](https://user-images.githubusercontent.com/106678594/210628851-079c7a86-50a8-47f3-9609-6e3b241ba9f0.png)

### Additional information
**Page Template**
These are Tier 1 Benefit Hub Landing Page templates, e.g. where you land from homepage Hub list links

<details><summary>Hub list homepage links</summary>

![image](https://user-images.githubusercontent.com/106678594/210629145-ddd48305-8c0e-497e-9ea6-61b54aa661d6.png)
</details>

Examples:
- /health-care
- /disability
- /education
- etc.

These pages have main content on the left and a second column of content on the right. e.g. https://www.va.gov/disability/

### **Proper heading structure**
Currently, the first promo block heading is an H4 heading, which is causing the issue.  However, if we simply change this to and H3 this would imply that this promo (and the subsequent accordion content) was underneath the previous H2, from a hierarchical standpoint which is not correct.
![image](https://user-images.githubusercontent.com/106678594/210630705-54e6369b-3cff-4afb-9a9a-a242c7c72072.png)

#### **NOTE**
Sometimes the Promo is not present in the right column - Example: https://www.va.gov/decision-reviews/
![image](https://user-images.githubusercontent.com/106678594/211100707-478158ac-fef0-402a-aca2-96673cf60738.png)

# Solution
![Screen Shot 2023-01-27 at 12 17 41 PM](https://user-images.githubusercontent.com/85581471/215189094-4ee5d043-3891-486c-8baf-0515bb6c2026.png)

**Promo block:**
* Make the header an H2, semantically, but presentation keeps h4
* Backlog ticket to unlink the header and create a field to manage the link & link text

**Accordions:**
* Change h3 accordions header to h2, keep visual presentation
* Change h4 to h3 headers semantically, keep visual presentation
* Backlog ticket to evaluate removing the right sidebar accordions

~~**On this page:**~~
~~* Include H2s, including Ask questions & Connect with us right sidebar H2s~~
~~* Do not include Promo block H2~~
During implementation, confirmed that On this page can handle H2s within `article`, sidebar is out of scope. If we want to make this change, we'll need a separate SPIKE / Implementation ticket to pursue it.

## Tasks

- [X] Content team intake for wording for new right sidebar H2 - https://github.com/department-of-veterans-affairs/va.gov-team/issues/52342

## AC / Expected behavior
Visual presentation of the page should not change in the right sidebar; only visual change is addition of new H2s to On this page. Other fixes are only semantic.

- [x] Confirm if this problem applies to any other templates with #hub-promo block atop the #hub-rail right sidebar
- [x] Assess if Promo block H4 > H3 change will impact any other templates
- [x] Hub promo block header is semantic H2 with visual H4 presentation. (e.g. presentation should not visually change
  - [x] Promo block used in megamenu header should not change.
- [x] Right sidebar accordions are visually unchanged. Headers are semantically updated: 
  - [x] Ask questions, Connect with us are semantic H2, visual H3
  - [x] Current H4s within right sidebar accordions are semantic H3, visual H4
~~On this page widget displays new Ask questions and Connect with us H2s~~
~~On this page widget _does not display_ the Promo box H2. If this poses an issue technically, raise a flag~~
- [ ] Accessibility review: @laflannery to review in Review Instance before merging
  - [ ] These changes should close https://github.com/department-of-veterans-affairs/va.gov-team/issues/9981

### Team
Please check the team(s) that will do this work.

- [ ] `CMS Team`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by jilladams** [comment]
@laflannery can you confirm for me this is what we mean here? 

![disability](https://user-images.githubusercontent.com/85581471/210892031-629cb40a-f66d-4469-b4e8-cd99e0573287.png)

**Comment by jilladams** [comment]
Also: I know Promo blocks are used in various places. Wouldn't be surprised if this is a global template that gets pulled into Benefit Hub Landing pages, so our H4 > H3 change may affect more than just these Benefit Hub Landing pages. We'll need to assess that -- I think we could check quickly during refinement, so not cutting a spike ticket for now, but have added to ACs.

**Comment by laflannery** [comment]
[Found a backlog team repo issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9981) that I will close when this is updated.

**Comment by jtmst** [comment]
@jilladams Has the copy been decided on for this yet?  If not, who would I reach out to to get that rolling?

**Comment by jilladams** [comment]
Ugh, lost track of that. We'll need to hit up the Content team. Creating ticket now. FYI @wesrowe

**Comment by jilladams** [comment]
Final decisions from meeting today with LauraF, LauraW, Matt Dingee, Jordan & Wes: 

Promo block:
* Make the header an H2, semantically, but presentation keeps h4
* Backlog ticket to unlink the header and create a field to manage the link & link text

Accordions:
* Change h3 accordions header to h2, keep visual presentation
* Change h4 to h3 headers semantically, keep visual presentation
* Backlog ticket to evaluate removing the right sidebar accordions

On this page:
* Include H2s, including Ask questions & Connect with us right sidebar H2s
* Do not include Promo block H2

Updating ticket / ACs.

**Comment by jilladams** [comment]
@jtmst when you get to this, holler if you want to discuss ACs or have any concerns. Specifically curious to confirm if semantic vs. visual changes and On this page widget changes are ok technically (include accordion header H2s, but not the promo box H2.)

**Comment by jtmst** [comment]
The 'on this page' component doesnt currently support H2s that arent located in the <article /> section of a page

**Comment by jilladams** [comment]
@jtmst does that mean we can't extend it to support H2s outside of `article` or that we need a SPIKE ticket to work out if / how / estimate?

**Comment by wesrowe** [comment]
I saw the slack thread and am okay with leaving sidebar out of On This Page.

**Comment by laflannery** [comment]
**Pa11y-ci report from yesterday that I generated prior to updates:**

- Axe runner
- WCAG2AA standard
- Looking for only heading violations on 11 Benefits landing pages (apparently missing one page when I ran this originally)

All pages failed:
![image](https://user-images.githubusercontent.com/106678594/216054934-7106ba13-f670-49cc-8951-9765c445a574.png)

**Reran the same scan this morning - except I included the one additional Landing page I missed yesterday:**

All reported heading issues for these pages are now resolved on production, 0 violations are reported:
![image](https://user-images.githubusercontent.com/106678594/216055409-1850da26-cc31-42d2-924c-89bdf3328019.png)
